components/tooltip/script.js

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Coherent Labs AD. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import components from 'coherent-gameface-components';
+import template from './template.html';
+const TOOLTIP_MARGIN = 10;
+
+class Tooltip extends HTMLElement {
+    constructor() {
+        super();
+        this.template = template;
+
+        this.state = {
+            visible: false
+        };
+    }
+
+    connectedCallback() {
+        this.position = this.getAttribute('position') || 'top';
+        this.showOn = this.getAttribute('on') || 'click';
+        this.hideOn = this.getAttribute('off') || 'click';
+        this.elementSelector = this.getAttribute('for');

```

Comment:
Changed for to target.

No need to validate the query selector as Chrome and Gameface already have built in in validation of the querySelector. It will not be OK to limit the selectors by checking for preceding symbols as you might want to show the tooltip on a custom element, in which case the selector will be a tag name. It's good practice to use unique selectors because what will happen if there are multiple elements with the same tag name, but the same can apply to class names. Adding such limitation implicitly will take away the freedom which the basicness of the components is meant to provide. It's the user's job to make sure the selectors are correct. They might add a valid id #myid which does not exist  - there's no way to guard against that - all we need is a valid querySelector.

Reply:
Fair enough^ -> "It's the user's job to make sure the selectors are correct."